src/Confluent.Kafka/Structs/Error.cs

```diff
@@ -23,14 +23,22 @@ public struct Error
         public Error(ErrorCode code)
         {
             Code = code;
+            String = null;
+        }
+
+        public Error(ErrorCode code, string reason)
```

Comment:
(not quite the right place for this comment, but close): I think `Error` would be better as a class (not a struct) for reasons mentioned elsewhere. Probably move Error.cs out of the structs folder as well.